gammapy/datasets/map.py

```diff
@@ -2571,17 +2575,37 @@ def compute_flux_psf_convolved(self):
 
         if self.model.spatial_model and not isinstance(self.geom, RegionGeom):
             value = value * self.compute_flux_spatial().quantity
+        else:
+            value = value * self.compute_flux_spatial().quantity
 
         if self.model.temporal_model:
             value *= self.compute_temporal_norm()
 
         return Map.from_geom(geom=self.geom, data=value.value, unit=value.unit)
 
     def _compute_flux_spatial(self):
-        """Compute spatial flux"""
-        value = self.model.spatial_model.integrate_geom(self.geom)
-        if self.psf and self.model.apply_irf["psf"]:
-            value = self.apply_psf(value)
+        """Compute spatial flux
+
+        Returns
+        ----------
+        value: `~astropy.units.Quantity`
+            Psf-corrected, integrated flux over a given region.
+        """
+        if isinstance(self.geom, RegionGeom):
+            wcs_geom = self.geom.to_wcs_geom()
+            values = self.model.spatial_model.integrate_geom(wcs_geom)
+            tmp_map = Map.from_geom(geom=wcs_geom, data=values.data,
+                                    unit=values.unit)
+
+            if self.psf and self.model.apply_irf["psf"]:
+                values = self.apply_psf(tmp_map)
+            value = ((values.data * wcs_geom.solid_angle())).sum()
```

Comment:
This misses the application of the region mask, that you had in your previous implementation. As you have called `.integrate_geom()` already, there is no need to multiply again with the solid angle...